MekHQ/src/mekhq/campaign/mission/resupplyAndCaches/Resupply.java

```diff
@@ -0,0 +1,2371 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MekHQ. If not, see <http://www.gnu.org/licenses/>.
+ */
+package mekhq.campaign.mission.resupplyAndCaches;
+
+import megamek.client.ui.swing.util.UIUtil;
+import megamek.common.Compute;
+import megamek.common.Entity;
+import megamek.common.Mek;
+import megamek.common.annotations.Nullable;
+import megamek.logging.MMLogger;
+import mekhq.MekHQ;
+import mekhq.campaign.Campaign;
+import mekhq.campaign.finances.Money;
+import mekhq.campaign.force.Force;
+import mekhq.campaign.force.StrategicFormation;
+import mekhq.campaign.icons.StandardForceIcon;
+import mekhq.campaign.market.procurement.Procurement;
+import mekhq.campaign.mission.AtBContract;
+import mekhq.campaign.mission.AtBDynamicScenario;
+import mekhq.campaign.mission.Loot;
+import mekhq.campaign.mission.ScenarioTemplate;
+import mekhq.campaign.mission.enums.AtBMoraleLevel;
+import mekhq.campaign.parts.*;
+import mekhq.campaign.parts.enums.PartQuality;
+import mekhq.campaign.parts.equipment.*;
+import mekhq.campaign.personnel.Person;
+import mekhq.campaign.personnel.Skill;
+import mekhq.campaign.personnel.SkillType;
+import mekhq.campaign.personnel.enums.PersonnelRole;
+import mekhq.campaign.stratcon.StratconCoords;
+import mekhq.campaign.stratcon.StratconScenario;
+import mekhq.campaign.stratcon.StratconTrackState;
+import mekhq.campaign.unit.Unit;
+import mekhq.campaign.universe.Faction;
+import mekhq.campaign.universe.Factions;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ActionListener;
+import java.awt.event.WindowAdapter;
+import java.awt.event.WindowEvent;
+import java.math.BigInteger;
+import java.util.List;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static java.lang.Math.round;
+import static megamek.common.MiscType.F_SPONSON_TURRET;
+import static megamek.common.enums.SkillLevel.NONE;
+import static mekhq.campaign.finances.enums.TransactionType.EQUIPMENT_PURCHASE;
+import static mekhq.campaign.market.procurement.Procurement.getFactionTechCode;
+import static mekhq.campaign.mission.enums.AtBMoraleLevel.CRITICAL;
+import static mekhq.campaign.mission.enums.AtBMoraleLevel.DOMINATING;
+import static mekhq.campaign.mission.enums.AtBMoraleLevel.STALEMATE;
+import static mekhq.campaign.stratcon.StratconContractInitializer.getUnoccupiedCoords;
+import static mekhq.campaign.stratcon.StratconRulesManager.generateExternalScenario;
+import static mekhq.campaign.stratcon.StratconRulesManager.getRandomTrack;
+import static mekhq.campaign.unit.Unit.getRandomUnitQuality;
+import static mekhq.campaign.universe.Factions.getFactionLogo;
+import static mekhq.utilities.ReportingUtilities.CLOSING_SPAN_TAG;
+import static mekhq.utilities.ReportingUtilities.spanOpeningWithCustomColor;
+
+public class Resupply {
```

Comment:
If possible, break most if not all the Panels/Jmenus/Etcs into their own classes.
Keep the data into a data object, lets call it "Ressuply", and have all the GUI be a class called RessuplyGUI.
The Ressuply GUI will interact with the Ressuply data through its interface, updating, increasing, reducing, etc. However, if I ever need to work with the "resupply" programatically I dont have to worry about a GUI poping out of nowhere.